src/auth/oauth2/service_account.rs

```diff
@@ -51,16 +54,18 @@ pub struct ServiceAccount {
 }
 
 impl ServiceAccount {
-    pub(crate) fn new(sa: credentials::ServiceAccount) -> Self {
-        Self {
+    pub(crate) fn try_new(sa: credentials::ServiceAccount) -> Result<Self, AuthBuilderError> {
+        let private_key = EncodingKey::from_rsa_pem(sa.private_key.as_bytes())?;
+        let token_uri = Uri::from_maybe_shared(sa.token_uri.clone())?;
+        Ok(Self {
             inner: Client::new(),
             header: header("JWT", sa.private_key_id),
-            private_key: EncodingKey::from_rsa_pem(sa.private_key.as_bytes()).unwrap(),
-            token_uri: Uri::from_maybe_shared(sa.token_uri.clone()).unwrap(),
```

Comment:
Avoiding these unwraps is critical for any client of this library who wants to propagate Encoding Key errors properly.

Currently, building `Credentials` from JSON succeeds as long as the `private_key` field is provided in the JSON, even if the value is not a valid RSA PEM key.

An alternative would be to catch and throw this error during the building of the `Credentials`, but I think it's reasonable to throw the error here.

Reply:
It's a stylistic choice, I'm fine to do it without declaring the variable